### ISSUE TITLE
Error parsing service account key json when passing secrets through a reusable workflow

### ISSUE THREAD

**Issue by stilljake** [google-github-actions/auth#299]
### TL;DR

When using the action in a reusable workflow which is called from another workflow. Authenticating with a  service account key seems to fail with an error that the credentials json could not be parsed. 

`Error: google-github-actions/auth failed with: retry function failed after 4 attempts: failed to parse service account key JSON credentials: unexpected token � in JSON at position 0`

The key file I am using is added as a repository secret and passed to the workflow via a caller workflow. When I use the action directly in the caller workflow, there is no issue.

### Expected behavior

Action should not fail to authenticate given the key file is added correctly as a secret

### Observed behavior

Job fails with the following error:

`Error: google-github-actions/auth failed with: retry function failed after 4 attempts: failed to parse service account key JSON credentials: unexpected token � in JSON at position 0`

### Action YAML

```yaml
Called workflow:

on:
  workflow_call:
    secrets:
      GCP_KEYFILE:
        required: true

jobs:
  called:
    runs-on: ubuntu-latest
      - name: 'Authenticate to Google Cloud'
        id: 'auth'
        uses: google-github-actions/auth@e8df18b60c5dd38ba618c121b779307266153fbf #v1.1.0
        with:
          credentials_json: ${{ secrets.GCP_KEYFILE }}


Caller workflow:

jobs:
  caller:
    uses: ./.github/workflows/reusable-workflow.yml
    secrets:
      GCP_KEYFILE: $ {{ secrets.GCP_KEYFILE }}
```

**Comment by stilljake** [comment]
After reading this back I noticed the error was a typo:

`GCP_KEYFILE: $ {{ secrets.GCP_KEYFILE }}`

space between $ and {{
